platform/core.windows/src/org/netbeans/core/windows/persistence/PersistenceManager.java

```diff
@@ -1364,6 +1363,16 @@ public static String escapeTcId4XmlContent (String tcName) {
         }
         return tcName;
     }
+
+    public ModeConfig createModeFromXml(String xml) throws IOException {
+        ModeParser modeParser = ModeParser.parseFromString("NO_NAME_YET", new HashSet(1));
```

Comment:
We used to add `// NOI18N` to mark strings that shouldn't be translated. Having them in some `static final String` constant would be even better, especially when used sever times.

`new HashSet(1)` seems strange. Such size cannot even access a single element (as the ratio is `0.75` by default and `1 * 0.75 < 1`), is that what you wanted?

Reply:
I copied `HashSet(1)` from the original code in WindowManagerParser:
```
    /**
     * Loads Mode configuration from the given file.
     * @param fo
     * @return
     * @throws IOException 
     * @since 2.70
     */
    public static ModeConfig loadModeConfigFrom( FileObject fo ) throws IOException {
        String modeName = fo.getName();
        ModeParser parser = new ModeParser(modeName, new HashSet(1));
        parser.setInLocalFolder(true);
        parser.setLocalParentFolder(fo.getParent());
        return parser.load();
    }
```
That was the only occurrence of calling that ctor without a Set of TopComponent names I think...

Reply:
Changing to static final and default HashSet ctor.